AmplifyPlugins/API/AWSAPICategoryPlugin/Support/Decode/GraphQLResponseDecoder+DecodeData.swift

```diff
@@ -28,10 +28,98 @@ extension GraphQLResponseDecoder {
         } else {
             let serializedJSON = try encoder.encode(graphQLData)
             let responseData = try decoder.decode(request.responseType, from: serializedJSON)
+            if responseData is Model {
```

Comment:
an object that is a Model will have metadata added to its array associations, and an object that is a ModelListMarker will be a List (not of type Model) so i don't think it will be both. I've refactored this to
1. check the graphQLData to make sure it is a Model that we can add metadata to 
2. check that the responseType == ModelListMarker.Type
by the refactor, there's no more decoding twice, and no necessary order of decoding